brave/brave-core
# Implement Search Result and New Tab Page ads attribution if ads are enabled or disabled

Resolves https://github.com/brave/brave-browser/issues/29888



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

If `BraveAds` feature is enabled and `should_always_run_service` is set to `true`:
- Confirm purchase intent, anti-targeting, subdivision-targeting, epsilon greedy bandit, issuers and catalog resources are only loaded if Brave Private Ads and/or Brave News are enabled.
- Confirm purchase intent, anti-targeting, subdivision-targeting, epsilon greedy bandit. issuers and catalog resources are unloaded if Brave Private Ads and Brave News are disabled.
- Confirm text classification and text embedding resources are only loaded if Brave Private Ads are enabled.
- Confirm text classification and text embedding resources are unloaded if Brave Private Ads is disabled.
- Confirm conversion resource is always loaded.
- Confirm ads are served after relaunching the browser.
- Confirm updated resource components are loaded.
- Confirm ads are served after gaining the users attention on desktop.
- Confirm ads are served periodically on mobile.
- Confirm conversions are working as expected.
- Confirm user activity is working as expected.
- Confirm ad events are working as expected, i.e. viewed, clicked etc. are sent to the server.
- Confirm users are only rewarded for ad impressions if they have enabled Brave Private Ads.
- Confirm search result ad events are triggered, such as viewed, clicked, conversion etc.
- Confirm history is  onlyadded to brave://rewards if Brave Private Ares are enabled

If `BraveAds` feature is enabled and `should_always_run_service` is set to `false`:
- Confirm purchase intent, anti-targeting, subdivision-targeting, epsilon greedy bandit, conversions, issuers and catalog resources are only loaded if Brave Private Ads and/or Brave News are enabled.
- Confirm text classification and text embedding resources are only loaded if Brave Private Ads are enabled.
- Confirm text classification and text embedding resources are unloaded if Brave Private Ads is disabled and Brave News is enabled.
- Confirm ads are served after relaunching the browser.
- Confirm updated resource components are loaded.
- Confirm ads are served after gaining the users attention on desktop.
- Confirm ads are served periodically on mobile.
- Confirm conversions are working as expected.
- Confirm user activity is working as expected.
- Confirm ad events are working as expected, i.e. viewed, clicked etc. are sent to the server.
- Confirm users are only rewarded for ad impressions if they have enabled Brave Private Ads.
- Confirm search result ad events are triggered, such as viewed, clicked, conversion etc.
- Confirm history is only added to brave://rewards if Brave Private Ares are enabled
